autogen/agentchat/contrib/vectordb/mongodb.py

```diff
@@ -0,0 +1,291 @@
+        if docs[0].get("id") is None:
+            raise ValueError("The document id is required.")
+        collection = self.get_collection(collection_name)
+        for doc in docs:
+            if "embedding" not in doc:
+                doc["embedding"] = np.array(self.embedding_function([str(doc["content"])])).tolist()[0]
+        if upsert:
+            for doc in docs:
+                return collection.replace_one({"id": doc["id"]}, doc, upsert=True)
+        else:
+            return collection.insert_many(docs)
+
+    def update_docs(self, docs: List[Document], collection_name: str = None):
+        """
+        Update documents in the collection of the vector database.
+
+        Args:
+            docs: List[Document] | A list of documents.
+            collection_name: str | The name of the collection. Default is None.
+        """
+        return self.insert_docs(docs, collection_name, upsert=True)
+
+    def delete_docs(self, ids: List[ItemID], collection_name: str = None):
+        """
+        Delete documents from the collection of the vector database.
+
+        Args:
+            ids: List[ItemID] | A list of document ids. Each id is a typed `ItemID`.
+            collection_name: str | The name of the collection. Default is None.
+        """
+        collection = self.get_collection(collection_name)
+        return collection.delete_many({"id": {"$in": ids}})
+
+    def get_docs_by_ids(self, ids: List[ItemID] = None, collection_name: str = None):
+        """
+        Retrieve documents from the collection of the vector database based on the ids.
+
+        Args:
+            ids: List[ItemID] | A list of document ids. If None, will return all the documents. Default is None.
+            collection_name: str | The name of the collection. Default is None.
+        """
+        results = []
+        if ids is None:
+            collection = self.get_collection(collection_name)
+            results = list(collection.find({}, {"embedding": 0}))
+        else:
+            for id in ids:
+                id = str(id)
+            collection = self.get_collection(collection_name)
+            results = list(collection.find({"id": {"$in": ids}}, {"embedding": 0}))
+        return results
+
+    def retrieve_docs(
+        self,
+        queries: List[str],
+        collection_name: str = None,
+        index_name: str = "default",
+        n_results: int = 10,
+        distance_threshold: float = -1,
+        **kwargs,
+    ) -> QueryResults:
+        """
+        Retrieve documents from the collection of the vector database based on the queries.
+
+        Args:
+            queries: List[str] | A list of queries. Each query is a string.
+            collection_name: str | The name of the collection. Default is None.
+            n_results: int | The number of relevant documents to return. Default is 10.
+            distance_threshold: float | The threshold for the distance score, only distance smaller than it will be
```

Comment:
Can you double check distance_threshold, it looks like it's actually similarity_threshold in the implementation. In the protocol, the larger the distance, the smaller the similarity. So distance smaller than distance_threshold will return items with higher similarity.

Reply:
I see what you mean -- Working on a fix.

Reply:
Atlas Vector Search assigns a score, in a fixed range from 0 to 1 only, to every document that it returns. For cosine and dotProduct [similarities](https://www.mongodb.com/docs/atlas/atlas-search/field-types/knn-vector/#std-label-fts-knn-vector-type-options), Atlas Vector Search normalizes the score using the following algorithm:

score = (1 + cosine/dot_product(v1,v2)) / 2

The vector score is already essentially vector distance, normalized between 0 and 1. 

What do you think is the best path forward here @thinkall ?
I changed the implementation to be:
```
pipeline.append({"$match": {"score": {"lte": distance_threshold}}})
```

instead of:
```
pipeline.append({"$match": {"score": {"gte": distance_threshold}}})
```

Reply:
In the signature, it's better to keep the same meaning of `distance_threshold` accross different vectordbs. In mongodb, I'd suggest you create a internal variable `similarity_threshold = 1 - distance_threshold`, and  do `pipeline.append({"$match": {"score": {"gte": similarity_threshold}}})`

Reply:
Don't forget to update test once you updated the code in mongodb.py
It's too late for me, I'll check again tomorrow. Thank you so much for your patience and your great contribution, @ranfysvalle02 . I believe we're almost ready to merge your wonderful PR.

Reply:
Got it! Thank you for your help @thinkall 
<img width="756" alt="Screenshot 2024-06-21 at 10 04 47 AM" src="https://github.com/microsoft/autogen/assets/6230773/1a6aeab8-ab92-4595-8f9a-f36a6f8cd04b">